.github/workflows/run_frameworks.yml

```diff
@@ -0,0 +1,65 @@
+name: Run Specific Frameworks
+
+# run either if *only* frameworks are edited, or always if the workflow or action is edited.
+on:
+  pull_request:
+    paths:
+      - 'frameworks/**'
+      - '!frameworks/shared/**'
+      - '!amlb/**'
+      - '!frameworks/shared/**'
+      - '!runbenchmark.py'
+      - '!requirements.txt'
+      - '.github/workflows/run_frameworks.yml'
+      - '.github/actions/runbenchmark/action.yml'
+
+
+jobs:
+  framework_changes:
+    name: Detect Changed AutoML Frameworks
+    runs-on: ubuntu-latest
+    outputs:
+      output1: ${{ steps.frameworks-diff.outputs.frameworks }}
+    steps:
+    - uses: actions/checkout@v2
+    - name: pull base branch
+      run: |
+        git fetch --all
+        git checkout -b $GITHUB_BASE_REF origin/$GITHUB_BASE_REF
+        git checkout $GITHUB_HEAD_REF
+    - name: store changed frameworks
+      id: frameworks-diff
+      run: |
+        echo "::set-output name=frameworks::`git diff --name-only HEAD..$GITHUB_BASE_REF | grep -o -i -P 'frameworks/(?!shared).*/' | uniq | sed -e 's/frameworks//' -e 's/\///g' | perl -p -e 's/\n//'`"
+
+
+  run_frameworks:
+    name: ${{ matrix.framework }}/${{ matrix.task }}
+    runs-on: ubuntu-latest
+    needs: framework_changes
+    strategy:
+      matrix:
+        framework: [autoxgboost, autogluon, gama, h2oautoml, mlplan, tpot, autosklearn]
+        task: [bioresponse, dresses-sales, eucalyptus, internet-advertisements, micro-mass, kc1, APSFailure]
+      fail-fast:  true
+
+    steps:
+      - uses: actions/checkout@v2
+        if: contains(needs.framework_changes.outputs.output1, matrix.framework)
+      - uses: actions/cache@v2
+        if: contains(needs.framework_changes.outputs.output1, matrix.framework)
+        with:
+          path: ~/.cache/pip
+          key: pip-${{ hashFiles('**/requirements.txt') }}
+          restore-keys: |
+            pip-
+      - uses: actions/setup-python@v2
+        if: contains(needs.framework_changes.outputs.output1, matrix.framework)
+        with:
+          python-version: '3.8'
+      - name: Run benchmark
+        if: contains(needs.framework_changes.outputs.output1, matrix.framework)
+        uses: ./.github/actions/runbenchmark
```

Comment:
is it still WIP?
how does it work? is output1 always a string or can there be more than one framework changed? not sure how `contains` work in this case.

Reply:
Values are cast to string, and `contains(search, item)` is `true` if `item` is contained in `search` ([doc](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#contains)).
`output1` is one string with all changed frameworks without separator (e.g. `autosklearnh2oautoml`).